Modules/vc-module-catalog/VirtoCommerce.CatalogModule.Test/ExportImport2Test.cs

```diff
@@ -0,0 +1,91 @@
+                var categorySearchResult = new CategorySearchResult { Results = searchResult.Results, TotalCount = searchResult.TotalCount };
+                return categorySearchResult;
+            }
+            );
+
+            var propertySearchServiceMock = new Mock<IPropertySearchService>();
+            propertySearchServiceMock.Setup(x => x.SearchPropertiesAsync(It.IsAny<PropertySearchCriteria>())).ReturnsAsync((PropertySearchCriteria y) =>
+            {
+                var searchResult = new string[] { "7", "8", "9", "10", "11", "12", "13", "14" }.EmulateSearch<Property>(y);
+                var propertySearchResult = new PropertySearchResult { Results = searchResult.Results, TotalCount = searchResult.TotalCount };
+                return propertySearchResult;
+            }
+            );
+
+            var propertyDictionaryItemSearchServiceMock = new Mock<IProperyDictionaryItemSearchService>();
+            propertyDictionaryItemSearchServiceMock.Setup(x => x.SearchAsync(It.IsAny<PropertyDictionaryItemSearchCriteria>())).ReturnsAsync((PropertyDictionaryItemSearchCriteria y) =>
+            {
+                var searchResult = new string[] { "15", "16", "17" }.EmulateSearch<PropertyDictionaryItem>(y);
+                var propertyDictionaryItemSearchResult = new PropertyDictionaryItemSearchResult { Results = searchResult.Results, TotalCount = searchResult.TotalCount };
+                return propertyDictionaryItemSearchResult;
+            }
+            );
+
+            var productSearchServiceMock = new Mock<IProductSearchService>();
+            productSearchServiceMock.Setup(x => x.SearchProductsAsync(It.IsAny<ProductSearchCriteria>())).ReturnsAsync((ProductSearchCriteria y) =>
+            {
+                var searchResult = new string[] { "18", "19", "20", "21", "22" }.EmulateSearch<CatalogProduct>(y);
+                var productSearchResult = new ProductSearchResult { Results = searchResult.Results, TotalCount = searchResult.TotalCount };
+                return productSearchResult;
+            }
+            );
+
+            var catalogDataQuery = new CatalogFullExportDataQuery();
+
+            var catalogDataSource = new CatalogFullExportPagedDataSource(
+                catalogSearchServiceMock.Object,
+                productSearchServiceMock.Object,
+                categorySearchServiceMock.Object,
+                propertySearchServiceMock.Object,
+                propertyDictionaryItemSearchServiceMock.Object,
+                catalogDataQuery
+                )
+            { PageSize = 5 };
+
+            var fetchResult = new List<IExportable>();
+            while (catalogDataSource.Fetch())
+            {
+                fetchResult.AddRange(catalogDataSource.Items);
+            }
+
+            Assert.Equal("1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22", string.Join(',', fetchResult.Select(x => x.Id)));
```

Comment:
Let's compare each assert.
This assert does not check count of pages and its content, only flow.